src/platform/utilities/feature-toggles/useFeatureToggle.js

```diff
@@ -1,27 +1,110 @@
-import { useSelector } from 'react-redux';
+import { useEffect, useMemo } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { setData } from '../../forms-system/src/js/actions';
 import TOGGLE_NAMES from './featureFlagNames';
 
 export const toggleValuesSelector = state => state.featureToggles || {};
 
 export const loadingTogglesSelector = state => state?.featureToggles?.loading;
 
-export const useFeatureToggle = () => {
-  const useToggleValue = toggleName => {
-    if (!Object.values(TOGGLE_NAMES).includes(toggleName)) {
-      throw new Error(
-        `Invalid toggle name "${toggleName}". Did you add the toggle name to featureFlagNames.js?`,
-      );
-    }
-    return useSelector(state => toggleValuesSelector(state)?.[toggleName]);
-  };
+/**
+ * Configuration type for a single feature toggle
+ * @typedef { Object } ToggleConfig
+ * @property { string } toggleName - CamelCased name of the feature toggle example: 'profileUseExperimental'
+ * @property { string } [formKey] - Optional custom key for form / session storage
+ */
 
-  const useToggleLoadingValue = () => {
-    return useSelector(state => loadingTogglesSelector(state));
-  };
+/**
+ * Get feature toggle loading state
+ * @returns {Boolean} - feature flag loading state
+ */
+export const useToggleLoadingValue = () => {
+  return useSelector(state => loadingTogglesSelector(state));
+};
+
+/**
+ * Hook to sync feature toggles with form data and session storage
+ * @param {(string|ToggleConfig)[]} toggles - Array of toggle name strings or ToggleConfig objects
+ * @param {Function} [setStorageItem] - Storage setter function (defaults to sessionStorage.setItem)
+ */
+export const useFormFeatureToggleSync = (
+  toggles,
+  setStorageItem = window.sessionStorage.setItem.bind(window.sessionStorage),
```

Comment:
if we bing the setItem in here as the default arg, then it makes it a bit easier to unit test later on.

Reply:
Nice!